Engine/entry1.c

```diff
@@ -135,6 +135,7 @@ OENTRY opcodlst_1[] = {
   { "ihold",  S(LINK),0,          "",     "",     ihold, NULL, NULL, NULL  },
   { "turnoff",S(LINK),0,           "",     "",     NULL,   turnoff, NULL, NULL },
   /* VL: 10.2.22 this was thread  but with parser3 we need to make string assignment on threads 1 & 2 */
+  { "=.instrS", S(IREF_NUM) ,0,  "S", ":InstrDef;", (SUBR) get_instr_name},
```

Comment:
Curious about this one. Seems odd to have an assignment overload here for InstrDef to String. One thing we might need moving forward from here is a convention for opcodes such as to_string(dataType) or str(datatype). So we could have:

S1 = str(myInstr)
S2 = to_string(myInstr)

(something like that).

Reply:
what about S()? Following the convention we already use.

Reply:
I think str() would be good, considering familiarity with python by a large part of our community. 

This would also be easy to overload for anyone introducing types, whether native or user-defined. S() seems odd to me and I didn't even know it existed. (I've always just used sprintf or some other formatted string opcode.) 

Rather than have a __str__ method as in Python, users could be directed to implement a UDO or opcode like:

```
struct MyNote start:i, dur:i, noteNum:i, veloc:i

opcode str(note:MyNote):S 
  return sprintf("[MyNote] %g %g %g %g", note.start, note.dur, note.noteNum, note.veloc)
endop
```

and the above could be a common pattern for user-defined types.